controller/network/network.go

```diff
@@ -265,7 +265,7 @@ func (network *Network) UnbindService(srcR *Router, token string, serviceId stri
 	return errors.New("invalid service")
 }
 
-func (network *Network) CreateSession(srcR *Router, clientId *identity.TokenId, serviceId string) (*session, error) {
+func (network *Network) CreateSession(srcR *Router, clientId *identity.TokenId, serviceId string, hints map[uint32][]byte) (*session, error) {
```

Comment:
Is there some way we can make this optional without having to pollute every usage of it with `nil` that isn't concerned with "hints"?

What is this used for?

Reply:
Like... couldn't we use `identity.TokenId` for this?

Reply:
`hints` is generic way of passing info between two endpoints. I intend to use that for key exchange. I wish go supported default parameter values, but no. We could duplicate the method but there was only one other use of it.

Reply:
I don't think "hints" is the right name for this concept.

Reply:
any suggestions?

Reply:
To me, this feels like something that should be pushed into `identity` somewhere, rather than done as a generic `map` in the fabric. If we had something like `identity.PeerKey` or similar, and that was part of the `CreateSession` call, that would make sense.

So yes... not into the name "hints", but even moreso, not really into just sticking a vague `map` here.

Reply:
A couple of points:
- I think we need a generic mechanism to pass peer info to the endpoint -- peer Id/app/version could be useful. We can even use it to implement transparent networking. 
- second, the key here is not really a peer key, it is a session key. Identity can have multiple sessions each starting with a new ephemeral key. Pushing ephemeral key support into identity framework does not make sense to me.

Reply:
`peerData` or `data` would probably be better. I just really don't like the name `hints`.

Also, you could certainly create your session, and then exchange your peer data. It doesn't necessarily _need_ to be layered into the initial session creation.

Reply:
This also might be a bit of premature optimization... sure having the fabric piggyback this data in the initial session creation might save you a couple of messages on the wire (you could do this as part of your protocol, above the fabric)... but as soon as you need to roll a key, you're going to have to do the key negotiation above the fabric anyway.

Reply:
seems that we can minimize impact with netfoundry/ziti-foundation#34